File: springvaadincrud/src/main/java/com/knf/dev/demo/springvaadincrud/backend/repository/UserRepository.java
package com.knf.dev.demo.springvaadincrud.backend.repository;

import com.knf.dev.demo.springvaadincrud.backend.model.User;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface UserRepository extends JpaRepository<User, Long> {

    List<User> findByEmailStartsWithIgnoreCase(String email);
}

File: kotlinspringvaadincrud/src/main/java/com/knf/dev/demo/kotlinspringvaadincrud/backend/repository/UserRepository.kt
package com.knf.dev.demo.kotlinspringvaadincrud.backend.repository

import com.knf.dev.demo.kotlinspringvaadincrud.backend.model.User
import org.springframework.data.jpa.repository.JpaRepository

interface UserRepository : JpaRepository<User?, Long?> {
    fun findByEmailStartsWithIgnoreCase(email: String?): List<User?>?
}